sound/soc/sof/sof-audio.c

```diff
@@ -40,6 +40,9 @@ int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 		dai->configured = false;
 	}
 
+	/* reset route setup status for all routes that contain this widget */
+	sof_reset_route_setup_status(sdev, swidget);
```

Comment:
and IPC3 just does not care?

Logically speaking it is a bad idea to free a component within an active route, right? Can this be a fixup to some earlier patch?